types/queries.go

```diff
@@ -359,3 +360,17 @@ type RawQuery struct {
 	ContractAddr string `json:"contract_addr"`
 	Key          []byte `json:"key"`
 }
+
+type ContractInfoQuery struct {
+	ContractAddr string `json:"contract_addr"`
+}
+
+type ContractInfoResponse struct {
+	CodeID  uint64 `json:"code_id"`
+	Creator string `json:"creator"`
+	// Set to the admin who can migrate contract, if any
+	Admin  string `json:"admin,omit_empty"`
+	Pinned bool   `json:"pinned"`
```

Comment:
Q: We have this info with an extra index in wasmd but I was wondering now if `pinned` would be useful for a contract? 🤷

Reply:
I don't know how you want to store it, but it is interesting info if you want an idea of gas cost (maybe you do powerful Smart Queries if pinned and weaker Raw Queries if not).